### PR TITLE
Resolve PoweredLightVisualizer is Obsolete

### PR DESCRIPTION
**Pull request by TemporalOroboros**
## About the PR

Splits the PoweredLightVisualizer into a component/system pair.

### TODO:
- [ ] Testing

**Media**

- [ ] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

### CODE REVIEW

File: Content.Client/Light/Visualizers/PoweredLightVisualizerComponent.cs [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+namespace Content.Client.Light.Visualizers;
+
+[RegisterComponent]
+[Access(typeof(PoweredLightVisualizerSystem))]
+public sealed class PoweredLightVisualizerComponent : Component
+{
+    [DataField("minBlinkingTime")] public float MinBlinkingTime = 0.5f;
+    [DataField("maxBlinkingTime")] public float MaxBlinkingTime = 2;
+    [DataField("blinkingSound")] public SoundSpecifier? BlinkingSound = default;
```
Comment (AJCM-git) [inline Content.Client/Light/Visualizers/PoweredLightVisualizerComponent.cs:11]: separate the attribute from the field (also VV)

File: Content.Client/Light/Visualizers/PoweredLightVisualizerComponent.cs [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+    public bool WasBlinking;
+
+    public Action<string>? BlinkingCallback;
```
Comment (AJCM-git) [inline Content.Client/Light/Visualizers/PoweredLightVisualizerComponent.cs:15]: VV?

File: Content.Client/Light/Visualizers/PoweredLightVisualizerSystem.cs [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+using Robust.Client.Animations;
+using Robust.Client.GameObjects;
+using Robust.Shared.Animations;
+using Robust.Shared.Audio;
+using Robust.Shared.Random;
+
+namespace Content.Client.Light.Visualizers;
+
+[RegisterComponent]
+public sealed class PoweredLightVisualizerComponent : Component
```
Comment (AJCM-git) [inline Content.Client/Light/Visualizers/PoweredLightVisualizerSystem.cs:12]: oops

File: Content.Client/Light/Visualizers/PoweredLightVisualizerSystem.cs [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+    {
+        base.Initialize();
+        SubscribeLocalEvent<PoweredLightVisualizerComponent, AnimationCompletedEvent>(OnAnimationCompleted);
+    }
+
+    protected override void OnAppearanceChange(EntityUid uid, PoweredLightVisualizerComponent comp, ref AppearanceChangeEvent args)
+    {
+        if (args.Sprite == null)
+            return;
+        if (!AppearanceSystem.TryGetData(uid, PoweredLightVisuals.BulbState, out PoweredLightState state, args.Component))
```
Comment (AJCM-git) [inline Content.Client/Light/Visualizers/PoweredLightVisualizerSystem.cs:42]: generic, newline

File: Content.Client/Light/Visualizers/PoweredLightVisualizerSystem.cs [hunk 5]
```diff
@@ -0,0 +1,147 @@
...
+                break;
+            case PoweredLightState.Broken:
+                args.Sprite.LayerSetState(PoweredLightLayers.Base, "broken");
+                ToggleBlinkingAnimation(uid, false, comp, ref args);
+                break;
+            case PoweredLightState.Burned:
+                args.Sprite.LayerSetState(PoweredLightLayers.Base, "burn");
+                ToggleBlinkingAnimation(uid, false, comp, ref args);
+                break;
+        }
```
Comment (AJCM-git) [inline Content.Client/Light/Visualizers/PoweredLightVisualizerSystem.cs:71]: unhardcode

File: Content.Client/Light/Visualizers/PoweredLightVisualizerSystem.cs [hunk 6]
```diff
@@ -0,0 +1,147 @@
...
+            case PoweredLightState.Off:
+                args.Sprite.LayerSetState(PoweredLightLayers.Base, "off");
+                ToggleBlinkingAnimation(uid, false, comp, ref args);
+                break;
+            case PoweredLightState.On:
+                if (AppearanceSystem.TryGetData(uid, PoweredLightVisuals.Blinking, out bool isBlinking, args.Component))
+                    ToggleBlinkingAnimation(uid, isBlinking, comp, ref args);
+                if (!isBlinking)
+                {
+                    args.Sprite!.LayerSetState(PoweredLightLayers.Base, "on");
```
Comment (AJCM-git) [inline Content.Client/Light/Visualizers/PoweredLightVisualizerSystem.cs:60]: i dont think the null forgiving operator here matters

File: Content.Client/Light/Visualizers/PoweredLightVisualizerSystem.cs [hunk 7]
```diff
@@ -0,0 +1,147 @@
...
+        var animationPlayer = EnsureComp<AnimationPlayerComponent>(uid);
+        if (isBlinking)
+        {
+            AnimationSystem.Play(uid, animationPlayer, BlinkingAnimation(comp), "blinking");
+        }
+        else if (AnimationSystem.HasRunningAnimation(uid, animationPlayer, "blinking"))
+        {
+            AnimationSystem.Stop(uid, animationPlayer, "blinking");
+        }
+    }
```
Comment (AJCM-git) [inline Content.Client/Light/Visualizers/PoweredLightVisualizerSystem.cs:99]: move the key to the component, i dont thunk `ToggleBlinkingAnimation` needs `AppearanceChangeEvent` as an argument, make `OnAnimationCompleted` use `ToggleBlinkingAnimation`

**Review by AJCM-git — CHANGES_REQUESTED** [review]

File: Content.Client/Light/Visualizers/PoweredLightVisualizerComponent.cs (hunk 2 above)
Comment (TemporalOroboros) [reply Content.Client/Light/Visualizers/PoweredLightVisualizerComponent.cs:15]: Removed

**Review by AJCM-git — APPROVED** [review]

**Pull request merged**